proposals/0000-local-import.rst

```diff
@@ -0,0 +1,167 @@
+Local import
+==============
+
+.. proposal-number:: Leave blank. This will be filled in when the proposal is
+                     accepted.
+.. trac-ticket:: Leave blank. This will eventually be filled with the Trac
+                 ticket number which will track the progress of the
+                 implementation of the feature.
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/0>`_.
+            **After creating the pull request, edit this file again, update the
+            number in the link, and delete this bold sentence.**
+.. sectnum::
+.. contents::
+
+We propose to allow the use of ``import`` statements in any ``let``, ``where`` or ``do`` block. The effect of those imports is limited to the corresponding block and (for ``let`` and ``where`` blocks) to the defined expression. Moreover, we define a shortcut syntax, ``Qualifier.{ … }``, to allow scoping a local import over a single expression.
```

Comment:
To avoid an inevitable question it might be worth mentioning that `Qualified.( … )` can't be used because it's already valid haskell and with a different meaning (I assume that it's the reason for the choice of the braces)

Reply:
:+1: